bin/check-http.rb

```diff
@@ -372,4 +389,28 @@ def handle_response(res, size, body)
       end
     end
   end
+
+  def apply_v4_signature_if_needed(req)
```

Comment:
could we move this to a library so it can be potentially reused in some of the other checks here?